src/webgpu/shader/execution/expression/call/builtin/textureSample.spec.ts

```diff
@@ -70,11 +81,74 @@ Parameters:
       Values outside of this range will result in a shader-creation error.
 `
   )
-  .paramsSubcasesOnly(u =>
+  .params(u =>
     u
-      .combine('S', ['clamp-to-edge', 'repeat', 'mirror-repeat'] as const)
-      .combine('coords', generateCoordBoundaries(2))
-      .combine('offset', generateOffsets(2))
+      .combine('format', kEncodableTextureFormats)
+      .filter(t => {
+        const type = kTextureFormatInfo[t.format].color?.type;
+        return type === 'float' || type === 'unfilterable-float';
+      })
+      .combine('sample_points', kSamplePointMethods)
+      .combine('addressModeU', ['clamp-to-edge', 'repeat', 'mirror-repeat'] as const)
+      .combine('addressModeV', ['clamp-to-edge', 'repeat', 'mirror-repeat'] as const)
+      .combine('minFilter', ['nearest', 'linear'] as const)
+      .combine('offset', [false, true] as const)
+  )
+  .beforeAllSubcases(t => {
+    const format = kTexelRepresentationInfo[t.params.format];
+    t.skipIfTextureFormatNotSupported(t.params.format);
+    const hasFloat32 = format.componentOrder.some(c => {
+      const info = format.componentInfo[c]!;
+      return info.dataType === 'float' && info.bitLength === 32;
+    });
+    if (hasFloat32) {
+      t.selectDeviceOrSkipTestCase('float32-filterable');
+    }
+  })
+  .fn(async t => {
+    const descriptor: GPUTextureDescriptor = {
+      format: t.params.format,
+      size: { width: 8, height: 8 },
+      usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.TEXTURE_BINDING,
+    };
+    const texelView = createRandomTexelView(descriptor);
+    const calls: TextureCall<vec2>[] = generateSamplePoints(50, t.params.minFilter === 'nearest', {
+      method: t.params.sample_points,
+      textureWidth: 8,
+      textureHeight: 8,
+    }).map((c, i) => {
+      const hash = hashU32(i) & 0xff;
+      return {
+        builtin: 'textureSample',
+        coordType: 'f',
+        coords: c,
+        offset: t.params.offset ? [(hash & 15) - 8, ((hash >> 8) & 15) - 8] : undefined,
```

Comment:
`hash >> 8` would always be 0, I think.
```suggestion
        offset: t.params.offset ? [(hash & 15) - 8, ((hash >> 4) & 15) - 8] : undefined,
```

Reply:
done